lib/stdlib/src/c.erl

```diff
@@ -689,17 +704,32 @@ display_info(Pid) ->
 	    iformat(w(Pid), mfa_string(Call),
 		    w(HS),
 		    w(Reds), w(LM)),
-	    iformat(case fetch(registered_name, Info) of
-			0 -> "";
-			X -> io_lib:format("~tw", [X])
-		    end,
+	    iformat(fetch_name(fetch(registered_name, Info), Info),
 		    mfa_string(Curr),
 		    w(SS),
 		    "",
 		    ""),
 	    {Reds, LM, HS, SS}
     end.
 
+fetch_name([], Info) ->
+    case fetch({dictionary, '$process_label'}, Info) of
+        undefined -> "";
+        Id -> format_name(Id)
+    end;
+fetch_name(Reg, _) ->
+    Reg.
+
+format_name(Id) when is_list(Id); is_binary(Id) ->
+    case unicode:characters_to_binary(Id) of
+        {error, _, _} ->
+            io_lib:format("~0.tp", [Id]);
+        BinString ->
+            BinString
+    end;
+format_name(TermId) ->
+    io_lib:format("~0.tp", [TermId]).
```

Comment:
this part is about fetching and formatting label. at a glance, `name` might suggested registered name.